### PR TITLE
Ticket/2180 blog category cache tag

### LINKED ISSUE #2180: Blog Category - Cache Issues
A Blog Category (Taxonomy Term) was updated, but the change did not apply to the Categories block on all the pages within the blog. (This was Cancer Currents)

I'm creating this issue because I:
1) had a tab open on https://www-prod-acsf.cancer.gov/news-events/cancer-currents-blog?topic=fda-approvals. The page title was "FDA Drug Approvals - Cancer Currents Blog" and the right rail category was "FDA Drug Approvals"
2) In another tab, I had www-cms.cancer.gov open and I changed the blog taxonomy term to remove 'Drug' so the category was "FDA Approvals"
3) On prod-ascf, I see my update when I'm on any of the other category pages (e.g., https://www-prod-acsf.cancer.gov/news-events/cancer-currents-blog?topic=disparities), but **not** when I'm on the fda-approvals page itself.

If this doesn't make sense, @bryanpizzillo can better explain/update this.

### LINKED ISSUE #2340: Refine Block and Cache Tags for Blog Archive Block
Currently the Blog Archive Block uses the node_list cache tag. This would be fired way too often for what is needed. We should use the blog series cache-tag instead (node:xxxx). Additionally the block should be rendered in such a way as to ensure that the cache tag is *actually* sent with all blog-posts that display the archive.

### LINKED ISSUE #2341: Refine Block and Cache Tags for Blog Featured Posts Block
Currently the Blog Featured Posts Block does not include any cache tags. This means that if a feature-post is added, blog-posts would not be invalidated from the cache. We should use the blog series cache-tag (node:xxxx) for the block. Additionally the block should be rendered in such a way as to ensure that the cache tag is actually sent with all blog-posts that display the archive.

### LINKED ISSUE #2342: Refine Block and Cache Tags for Blog Pager Block
Currently the Blog Pager Block (for older and newer posts) uses the node_list cache tag. This would be fired way too often for what is needed. We should use the blog series cache-tag instead (node:xxxx). Additionally the block should be rendered in such a way as to ensure that the cache tag is actually sent with all blog-posts that display the archive.

### PR DESCRIPTION
**Pull request by kalidindir**
Closes #2180 
Closes #2340 
Closes #2341 
Closes #2342

### CODE REVIEW

**Comment by VictoriaSunNIH** [comment]
ODE: http://ncigovcdode256.prod.acquia-sites.com

**Comment by kalidindir** [comment]
Closes #2340 
Closes #2341 
Closes #2342

**Comment by lburack** [comment]
@kalidindir - All the tickets associated with this PR look good to me. Comments are on the individual tickets, but 
- @VictoriaSunNIH  and I both tested changing blog category names in #2180 
- I tested adding and removing posts from the archive in #2340 
- I tested changing titles, adding, and removing Featured Posts in #2341
- #2342 - I'm not sure exactly how to test this. In the blog series, I changed the number of posts to display to 150. Everything seemed okay. I went back to displaying 10 posts and everything seemed to check out again. @bryanpizzillo  @kalidindir - Let me know if there's something else I should be doing with that one.

**Comment by bryanpizzillo** [comment]
There is an issue with Categories:
1. Create a new blog series
2. Create a post for that series
3. View that post in incognito window - keep the window open
4. Create a new topic for the series (it will be the first)
5. Create a new post and tag it to the topic in step 4.
6. View the second post -- notice the categories block
7. Refresh the incognito window with the first post. Notice the block is missing.

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_blog/src/Plugin/Block/BlogCategories.php [hunk 1]
```diff
@@ -72,17 +74,33 @@ public function build() {
...
+  /**
+   * {@inheritdoc}
+   */
+  protected function blockAccess(AccountInterface $account) {
+    $blog_categories = $this->drawBlogCategories();
+    if (count($blog_categories) > 0) {
+      return AccessResult::allowed();
+    }
+    return AccessResult::forbidden();
+  }
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_blog/src/Plugin/Block/BlogCategories.php]: There needs to be a cache tag added to the AccessResult. As it is not set the block will not clear under the scenario defined in https://github.com/NCIOCPL/cgov-digital-platform/pull/2346#issuecomment-524038684.

**Pull request merged**